### PR TITLE
Implement deploy template

### PR DESCRIPTION
**Pull request by jpbulloch5**
Implements the template on: https://github.com/1053-August-Duet-Project-Registry/project-registry-devops/pull/12

Refactored to use deployment jobs and to handle conditions outside of the template.  Deployments execute only after pull-requests to branch-dev or main are approved.

### CODE REVIEW

**Review by escalonn — CHANGES_REQUESTED** [review]
https://github.com/1053-August-Duet-Project-Registry/project-registry-gateway/pull/8#pullrequestreview-715500630

File: devops/azure-pipelines.yml [hunk 1]
```diff
@@ -76,8 +76,55 @@ stages:
         containerServiceConnection: 'acr-team1'
         discordChannel: $(discordFrontendChannel)
         discordKey: $(discordFrontendKey)
-
         workDir: "./"
 
     displayName: Dockerize frontend --> Job
   displayName: Dockerize frontend --> Stage 
+
+- stage: deploy
```
Comment (escalonn) [inline devops/azure-pipelines.yml:84]: put a condition here that doesn't consider deploying if it's a pullrequest build. i think if there was a PR from dev, sourcebranch might still say dev.

**Review by escalonn — DISMISSED** [review]

File: devops/azure-pipelines.yml (hunk 1 above)
Comment (jpbulloch5) [reply devops/azure-pipelines.yml:84]: Oops, I have this on the Gateway and forgot to copy it to the others.

Comment (escalonn) [reply devops/azure-pipelines.yml:84]: i'll just remove my request for review from the others then, so re-add it when fixed.

Comment (jpbulloch5) [reply devops/azure-pipelines.yml:84]: Okay, I got them fixed.

**Review by escalonn — DISMISSED** [review]

**Review by escalonn — APPROVED** [review]

**Review by BeoHawk25 — APPROVED** [review]

**Pull request merged**